### ISSUE TITLE
5 test fail because of the same code

### ISSUE THREAD

**Issue by euri10** [pipermerriam/populus#88]
* Populus Version: 0.8.0
* OS: linux

### What was wrong?
```
================================================================================= ERRORS =================================================================================
_________________________________________________ ERROR at setup of test_contract_return13_function_signature[ethtester] _________________________________________________
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/populus/tests/contracts/test_contract_object.py", line 26, in deployed_math
    Math,
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/contracts/utils.py", line 80, in deploy_contract
    txn_hash = rpc_client.send_transaction(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/ethtester_client.py", line 222, in send_transaction
    raise ValueError("Timeout waiting for {0}".format(request_id))
ValueError: Timeout waiting for 3eee5eb1-9eb3-4558-9c07-1a8b64d10add
------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------
Resetting EVM state...
___________________________________________________ ERROR at setup of test_contract_add_function_signature[ethtester] ____________________________________________________
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/populus/tests/contracts/test_contract_object.py", line 26, in deployed_math
    Math,
  File "/home/lotso/PycharmProjects/populus/tests/contracts/test_contract_object.py", line 26, in deployed_math                                                 [106/1488]
    Math,
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/contracts/utils.py", line 80, in deploy_contract
    txn_hash = rpc_client.send_transaction(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/ethtester_client.py", line 222, in send_transaction
    raise ValueError("Timeout waiting for {0}".format(request_id))
ValueError: Timeout waiting for 07406287-f52f-46ec-b661-4e1fb26ec8f4
------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------
Resetting EVM state...
________________________________________________ ERROR at setup of test_contract_multiply7_function_signature[ethtester] _________________________________________________
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/populus/tests/contracts/test_contract_object.py", line 26, in deployed_math
    Math,
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/contracts/utils.py", line 80, in deploy_contract
    txn_hash = rpc_client.send_transaction(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/ethtester_client.py", line 222, in send_transaction
    raise ValueError("Timeout waiting for {0}".format(request_id))
ValueError: Timeout waiting for a5b51918-52f3-4ab6-ab4c-cc88e73ad87c
------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------
Resetting EVM state...
___________________________________________________ ERROR at setup of test_contract_function_call_return13[ethtester] ____________________________________________________
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/populus/tests/contracts/test_contract_object.py", line 26, in deployed_math
    Math,
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/contracts/utils.py", line 80, in deploy_contract
    txn_hash = rpc_client.send_transaction(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/ethtester_client.py", line 222, in send_transaction
    raise ValueError("Timeout waiting for {0}".format(request_id))
ValueError: Timeout waiting for 67109b77-7fc6-4f56-ae6a-3630eb271a08
------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------
Resetting EVM state...
___________________________________________________ ERROR at setup of test_contract_function_call_multiply7[ethtester] ___________________________________________________
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/populus/tests/contracts/test_contract_object.py", line 26, in deployed_math
    Math,
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/contracts/utils.py", line 80, in deploy_contract
    txn_hash = rpc_client.send_transaction(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/ethtester_client.py", line 222, in send_transaction
    raise ValueError("Timeout waiting for {0}".format(request_id))
ValueError: Timeout waiting for 4270ec02-5304-4647-8841-0d7e49ec3448
------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------
Resetting EVM state...
================================================================================ FAILURES ================================================================================
________________________________________________________________ test_deployment_with_no_args[ethtester] _________________________________________________________________
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/populus/tests/contracts/test_contract_deployment.py", line 20, in test_deployment_with_no_args
    Math,
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/contracts/utils.py", line 80, in deploy_contract
    txn_hash = rpc_client.send_transaction(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/ethtester_client.py", line 222, in send_transaction
    raise ValueError("Timeout waiting for {0}".format(request_id))
ValueError: Timeout waiting for dc1b261d-9db6-4ec9-bef0-8db4c9c840f1
------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------
Resetting EVM state...
Resetting EVM state...                                                                                                                                           [53/1488]
_______________________________________________________________ test_deployment_with_endowment[ethtester] ________________________________________________________________
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/populus/tests/contracts/test_contract_deployment.py", line 30, in test_deployment_with_endowment
    value=1000,
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/contracts/utils.py", line 80, in deploy_contract
    txn_hash = rpc_client.send_transaction(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/ethtester_client.py", line 222, in send_transaction
    raise ValueError("Timeout waiting for {0}".format(request_id))
ValueError: Timeout waiting for d9f0c86f-1b12-4afa-9398-7348298cf60d
------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------
Resetting EVM state...
__________________________________________________________________ test_deployment_with_args[ethtester] __________________________________________________________________
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/populus/tests/contracts/test_contract_deployment.py", line 42, in test_deployment_with_args
    constructor_args=("John",),
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/contracts/utils.py", line 80, in deploy_contract
    txn_hash = rpc_client.send_transaction(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/ethtester_client.py", line 222, in send_transaction
    raise ValueError("Timeout waiting for {0}".format(request_id))
ValueError: Timeout waiting for 7a7615e6-a62a-489d-89b0-ac80c3e1237b
------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------
Resetting EVM state...
___________________________________________________________________ test_double_index_event[ethtester] ___________________________________________________________________
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/populus/tests/contracts/test_event_logging.py", line 38, in test_double_index_event
    txn_a_hash = deployed_logs_events.logDoubleIndex('test-key-a', 'test-key-b', 'test-val_a', 12345)
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/contracts/functions.py", line 58, in __call__
    return self.sendTransaction(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/contracts/functions.py", line 88, in sendTransaction
    **kwargs
  File "/usr/local/lib/python2.7/dist-packages/populus-0.8.0-py2.7.egg/populus/ethtester_client.py", line 222, in send_transaction
    raise ValueError("Timeout waiting for {0}".format(request_id))
ValueError: Timeout waiting for e4471269-b14f-4bb7-9128-3b7280947625
------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------
Resetting EVM state...
__________________________________________________________________________ test_async_requests ___________________________________________________________________________
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/populus/tests/ethtester/test_ethtester_async.py", line 36, in test_async_requests
    assert not errors
AssertionError: assert not [ValueError('Timeout waiting for 404d2609-7545-404c-a8d8-3387a44070f2',), ValueError('Timeout waiting for 88da010f-1b1...g for 74127f4f-d499-43
52-9d93-4536e657ee9c',), ValueError('Timeout waiting for b0ee685e-8441-45c5-b0b6-e98f56d05fb9',)]
-------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------
Exception in thread Thread-58:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/lotso/PycharmProjects/populus/tests/ethtester/test_ethtester_async.py", line 25, in spam_block_number
    pytest.fail(e.message)
  File "/usr/lib/python2.7/dist-packages/_pytest/runner.py", line 486, in fail
    raise Failed(msg=msg, pytrace=pytrace)
Failed: Timeout waiting for 88da010f-1b12-4ed3-88f8-dd2586dbe521
Failed: Timeout waiting for 88da010f-1b12-4ed3-88f8-dd2586dbe521                                                                                                  [0/1488]

Exception in thread Thread-60:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/lotso/PycharmProjects/populus/tests/ethtester/test_ethtester_async.py", line 25, in spam_block_number
    pytest.fail(e.message)
  File "/usr/lib/python2.7/dist-packages/_pytest/runner.py", line 486, in fail
    raise Failed(msg=msg, pytrace=pytrace)
Failed: Timeout waiting for 74127f4f-d499-4352-9d93-4536e657ee9c

Exception in thread Thread-57:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/lotso/PycharmProjects/populus/tests/ethtester/test_ethtester_async.py", line 25, in spam_block_number
    pytest.fail(e.message)
  File "/usr/lib/python2.7/dist-packages/_pytest/runner.py", line 486, in fail
    raise Failed(msg=msg, pytrace=pytrace)
Failed: Timeout waiting for 1fb36143-409c-44b8-a912-806c32ff9fd1

Exception in thread Thread-59:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/lotso/PycharmProjects/populus/tests/ethtester/test_ethtester_async.py", line 25, in spam_block_number
    pytest.fail(e.message)
  File "/usr/lib/python2.7/dist-packages/_pytest/runner.py", line 486, in fail
    raise Failed(msg=msg, pytrace=pytrace)
Failed: Timeout waiting for 404d2609-7545-404c-a8d8-3387a44070f2

Exception in thread Thread-56:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/lotso/PycharmProjects/populus/tests/ethtester/test_ethtester_async.py", line 25, in spam_block_number
    pytest.fail(e.message)
  File "/usr/lib/python2.7/dist-packages/_pytest/runner.py", line 486, in fail
    raise Failed(msg=msg, pytrace=pytrace)
Failed: Timeout waiting for b0ee685e-8441-45c5-b0b6-e98f56d05fb9


```

#### Cute Animal Picture

![cute](http://petupon.com/wp-content/uploads/2014/05/259758xcitefun-cute-animals-0.jpg
)

**Comment by euri10** [comment]
can't explain it, but this diff solves my issue and all tests pass fine with it, I have absoletely NO idea  to why
```
diff --git a/populus/ethtester_client.py b/populus/ethtester_client.py
index 725ac47..42451f7 100644
--- a/populus/ethtester_client.py
+++ b/populus/ethtester_client.py
@@ -214,6 +214,7 @@ class EthTesterClient(object):
             self.request_queue.put((request_id, args, kwargs))
             start = time.time()
             while time.time() - start < self.async_timeout:
+                print('request_id: {} | self.results: {}'.format(request_id, self.results))
                 if request_id in self.results:
                     result = self.results.pop(request_id)
                     if isinstance(result, Exception):
```

**Comment by pipermerriam** [comment]
That is excellently weird.  Looking into it.

**Comment by euri10** [comment]
weird indeed,
the test fails for me when I tested yesterday from work through ssh on my home computer.
now sitting at home the test pass in like 39s
adding the print line, the test pass in like 2s
there seem to be some lock somewhere that timeout and raise the exception that the print unlocks for some reason.
I got no idea where to look at

**Comment by pipermerriam** [comment]
I'm very confused by this as well.  Will try to carve out a little time
again this morning to look this over.

**Comment by pipermerriam** [comment]
This should be fixed once #105  is merged.
